SU2_CFD/src/solvers/CSolverFactory.cpp

```diff
@@ -394,13 +395,25 @@ CSolver* CSolverFactory::CreateSpeciesSolver(CSolver **solver, CGeometry *geomet
 
   CSolver *speciesSolver = nullptr;
 
-  if (config->GetKind_Species_Model() != SPECIES_MODEL::NONE) {
-    if (adjoint){
-      speciesSolver = new CDiscAdjSolver(geometry, config, solver[SPECIES_SOL], RUNTIME_SPECIES_SYS, iMGLevel);
-    } else {
-      speciesSolver = new CSpeciesSolver(geometry, config, iMGLevel);
```

Comment:
Only the primal needs to change for flamelet. Avoid duplicating blocks of code please.